src/Raven.Server/Documents/Sharding/Handlers/Processors/Batches/ShardedClusterTransactionRequestProcessor.cs

```diff
@@ -27,12 +28,13 @@ public ShardedClusterTransactionRequestProcessor(ShardedDatabaseRequestHandler r
 
     public override Task WaitForDatabaseCompletion(Task<HashSet<string>> onDatabaseCompletionTask, long index, ClusterTransactionOptions options, CancellationToken token)
     {
-        return Task.CompletedTask;
+        var op = new ShardedWaitForIndexNotificationOperation(RequestHandler, index);
+        return RequestHandler.ShardExecutor.ExecuteParallelForAllAsync(op, token);
```

Comment:
use `WaitForExecutionOnShardsAsync(index)` instead

Reply:
note, that we need to add a token to that method

Reply:
Done